### DISCUSSION TITLE
How to combine Tests from all the classes into one XML execution file.

### DISCUSSION THREAD

**Discussion by shrubom** [Xray-App/xray-junit-extensions#48] [Q&A] [answered]
Hi @bitcoder ,

I have my tests for regression in different classes and they all link to different Test issue key. 
However, XrayTestAnnotations still only seem to pull for one test class. 
Can you help me with this issue?

**Example:**
FirstTest.class -> has 3 Tests that are linking to different Test issue keys in jira.
SecondTest.class -> has 2 Tests that are linking to 2 other Test issue keys in Jira

However, my Junit Test execution which Xray uses is still one. 
**Example:** TEST-skeleton-project.xml

However, It does not seem to pull the tests from the SecondTest.class. It completely ignores it.

**Jenkins import stage:**
stage('Xray Import'){
            steps{
                script{
                   def xrayStuff = step([$class: 'XrayImportBuilder', endpointName: '/junit', importFilePath: 'target/TEST-skeleton-project.xml', importInParallel: 'false', importToSameExecution: 'true', projectKey: 'TEST',
                   serverInstance: 'TEST', testEnvironments: 'QA'])
                }
            }
        }

Please advice. 

Thank you.

**Comment by bitcoder** [comment]
I think this may be related with  #47 ... I'll need to check when I have some time

**Reply by shrubom** [reply]
Yes yes yes. It is overwriting the previous one.

**Reply by bitcoder** [reply]
Can you provide some basic sample project code that shows this behavior as I was unable to replicate it?

**Reply by bitcoder** [reply]
I was able to replicate it, no need for a sample code

**Comment by bitcoder** [comment]
Please see my answer on #47 . I think using a recent version of surefire will solve your issue

**Comment by shrubom** [comment]
Will try it and keep you posted. 👍🏻

**Comment by shrubom** [comment]
Hi @bitcoder , 

I updated the maven surefire plugin to the latest version(3.2.5). The test execution report which is used for Xray Imports is no longer being generated. 
![image](https://github.com/Xray-App/xray-junit-extensions/assets/44477101/9c9cd764-8448-40bb-8824-b3a6c55b202d)

I added the plugin for xray-maven-plugin under the build section as well still no luck. 
![image](https://github.com/Xray-App/xray-junit-extensions/assets/44477101/66448f8d-6b87-48b6-8a9f-2a3445f7f499)

xray-junit-extensions.properties file:
report_filename=TEST-skeleton-project
report_directory=target
add_timestamp_to_report_filename=true
report_only_annotated_tests=false

Could you please check again.

**Reply by bitcoder** [reply]
Can you  provide a very basic project with dummy tests just to replicate it? 
Have you enabled the listener btw, using the  src/test/META-INF/services/org.junit.platform.launcher.TestExecutionListener. ?

**Comment by shrubom** [comment]
I wrote a completely simple Junit5 test with latest dependency and it works perfectly fine. So the reason it does not work in my project is probably because of dependency conflicts from the libraries I use. 

![image](https://github.com/Xray-App/xray-junit-extensions/assets/44477101/d80b907b-d164-45f7-bd63-500d89490789)

**Reply by bitcoder** [reply]
Great

**Reply by shrubom** [reply, marked as answer]
After fixing the dependency conflicts now they work as expected. Thank you so much.
